test/functional/p2p_disconnect_ban.py

```diff
@@ -102,7 +102,9 @@ def run_test(self):
                 assert_equal(ban["ban_duration"], 120)
                 assert_equal(ban["time_remaining"], 117)
 
-        self.restart_node(1)
+        # Keep mocktime, to avoid ban expiry when restart takes longer than
+        # time_remaining
+        self.restart_node(1, extra_args=[f"-mocktime={old_time+4}"])
```

Comment:
At first glance, this seems like it may allow for node 0 and node 1 to diverge in time (node 1's time is forced to a point potentially in the past, while node 0 isn't).  Had an initial thought about the ramifications of allowing this to happen.  Since this test is exercising banning rather than block generation, I'm not sure this would matter.

Reply:
There's the "maximum 2-hour block timestamp difference" acceptance rule, but I don't think Valgrind is *that* slow 😆